src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -0,0 +1,231 @@
+            Page(index, imageUrl = imageUrl)
+        }
+    }
+
+    override fun popularMangaFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.htmlDocumentToDto()
+        return MangasPage(
+            mangas = dto.toPopularMangaModel(),
+            hasNextPage = dto.hasNextPageToPopularMangas(),
+        )
+    }
+
+    override fun popularMangaNextPageSelector() = null
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/${langOption.infix}")
+
+    override fun popularMangaSelector() = "main > div.pt-1 > a"
+
+    override fun searchMangaFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun searchMangaNextPageSelector() = ""
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val maxResult = 6
+        val pathSearch = when (langOption.lang) {
+            "it" -> langOption.infix
+            else -> "v3/po"
+        }
+        val url = "$apiUrl/api/$pathSearch/searchforms/$maxResult/".toHttpUrl().newBuilder()
+            .addPathSegment(query)
+            .addPathSegment("${page - 1}")
+            .build()
+        return GET(url, apiHeaders(url.toString()))
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.startsWith(slugPrefix)) {
+            val mangaUrl = query.substringAfter(slugPrefix)
+            return client.newCall(GET("$baseUrl/${langOption.infix}/$mangaUrl", headers))
+                .asObservableSuccess().map { response ->
+                    val manga = mangaDetailsParse(response).apply {
+                        url = mangaUrl
+                    }
+                    MangasPage(listOf(manga), false)
+                }
+        }
+        return super.fetchSearchManga(page, query, filters)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val mangasDto = response.toMangaPageDto()
+        return MangasPage(
+            mangas = mangasDto.toModels(langOption.infix),
+            hasNextPage = mangasDto.hasNextPage(),
+        )
+    }
+
+    override fun searchMangaSelector() = ""
+
+    private fun Response.htmlDocumentToDto(): UnionMangasDto {
+        val jsonContent = asJsoup().selectFirst("script#__NEXT_DATA__")!!.html()
+        return json.decodeFromString<UnionMangasDto>(jsonContent)
+    }
+
+    private fun Document.htmlDocumentToDto(): UnionMangasDto {
+        val jsonContent = selectFirst("script#__NEXT_DATA__")!!.html()
+        return json.decodeFromString<UnionMangasDto>(jsonContent)
+    }
+
+    private fun Response.toMangaPageDto() = json.decodeFromString<PageableMangaListDto>(body.string())
+
+    private fun Response.toChapterPageDto() = json.decodeFromString<ChapterPageDto>(body.string())
```

Comment:
these seem unnecessary
```
private inline fun <reified T> Response.parseAs(): T = use {
        json.decodeFromString(body.string())
}
```
which you can use as: `response.parseAs<ChapterPageDto>()`